etc/inc/dyndns.class

```diff
@@ -695,8 +695,14 @@
 						log_error("GratisDNS.dk ({$this->_dnsHost}): DNS update() starting.");
 					}
 					$server = "https://ssl.gratisdns.dk/ddns.phtml";
-					list($hostname, $domain) = explode(".", $this->_dnsHost, 2);
-					curl_setopt($ch, CURLOPT_URL, $server . '?u=' . $this->_dnsUser . '&p=' . $this->_dnsPass . '&h=' . $this->_dnsHost . '&d=' . $domain . '&i=' . $this->_dnsIP);
+					$host = trim($this->_dnsHost);
+					$hostnames = explode(".", $host);
+					$hostnames_count = count($hostnames);
```

Comment:
You're right, it should be > 2.

GratisDNS expects:
www.bob.dk : h=www.bob.dk d=bob.dk
bob.dk : h=bob.dk d=bob.dk